### PR TITLE
fix, Distinguish between available models and models for evaluation so not all are reported

### PR DESCRIPTION
**Pull request by zimmski**

### CODE REVIEW

**Comment by zimmski** [comment]
Early merge of https://github.com/symflower/eval-dev-quality/pull/53

File: cmd/eval-dev-quality/cmd/evaluate_test.go [hunk 1]
```diff
@@ -54,7 +57,10 @@ func TestEvaluateExecute(t *testing.T) {
 			"--repository", "golang/plain",
 		},
 
-		ExpectedOutputContains: `Evaluation score for "symflower/symbolic-execution" ("code-no-excess"): score=6, coverage-statement=1, files-executed=1, response-no-error=1, response-no-excess=1, response-not-empty=1, response-with-code=1`,
+		ExpectedOutputValidate: func(t *testing.T, output string, resultPath string) {
+			assert.Contains(t, output, `Evaluation score for "symflower/symbolic-execution" ("code-no-excess"): score=6, coverage-statement=1, files-executed=1, response-no-error=1, response-no-excess=1, response-not-empty=1, response-with-code=1`)
+			assert.Equal(t, 1, strings.Count(output, "Evaluation score for"))
```
Comment (zimmski) [inline cmd/eval-dev-quality/cmd/evaluate_test.go:62]: Let's log the output if this fails for debugging.

File: cmd/eval-dev-quality/cmd/evaluate_test.go [hunk 2]
```diff
@@ -54,7 +57,10 @@ func TestEvaluateExecute(t *testing.T) {
...
+		},
 		ExpectedResultFiles: []string{
 			"evaluation.csv",
```
Comment (zimmski) [inline cmd/eval-dev-quality/cmd/evaluate_test.go:65]: Will check the CSV as well

**Review by bauersimon — CHANGES_REQUESTED** [review]

File: cmd/eval-dev-quality/cmd/evaluate.go [hunk 3]
```diff
@@ -59,33 +59,45 @@ func (command *Evaluate) Execute(args []string) (err error) {
...
-		languages = language.Languages
-	} else {
-		for _, languageID := range command.Languages {
-			l, ok := language.Languages[languageID]
-			if !ok {
-				ls := maps.Keys(language.Languages)
-				sort.Strings(ls)
+	languagesSelected := map[string]language.Language{}
+	{
+		languages := map[string]language.Language{}
```
[the remark this answers is not in the document]
Comment (zimmski) [reply cmd/eval-dev-quality/cmd/evaluate.go:64]: The "languages" map is used for "all languages" for the error messages. To make that clear i prefer to have a second map TBH. Makes is have one consistent behavior.

**Review by bauersimon — APPROVED** [review]

**Pull request merged**